### PR TITLE
Addon essentials

### PR DESCRIPTION
**Pull request by shilman**
Issue: #5731

## What I did

- [x] Add a preset to `addon-knobs` that automatically installs its decorator (unless disabled)
- [x] Add `addon-essentials`, a curated set of addons:
  - w/o presets: `actions`, `backgrounds`, `links`, `viewport`
  - w/ presets: `knobs`, `docs` (to be removed before release)
- [x] Add a sample app `cra-ts-essentials` that demonstrates out-of-box behavior
- [x] Warn to let users know that there may be overlapping configuration if any of  its addons are detected in their app

## How to test

See the `addon-essentials` README for how to configure, and `cra-ts-essentials` for a test environment.

### CODE REVIEW

File: examples/cra-ts-essentials/.storybook/config.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { configure } from '@storybook/react';
+
+// automatically import all files ending in *.stories.(tsx|jsx)
+configure(require.context('../src/stories', true, /\.stories\.[tj]sx?$/), module);
```
Comment (tmeasday) [inline examples/cra-ts-essentials/.storybook/config.ts]: Let's do a triconfig setup?

**Review by tmeasday — COMMENTED** [review]
LGTM with one change

**Comment by shilman** [comment]
@ndelangen @tmeasday I tried to upgrade `cra-ts-essentials` to triconfig by replacing the existing config/presets with `main.js`:

```
module.exports = {
  presets: [
    '@storybook/preset-create-react-app',
    {
      name: '@storybook/addon-essentials',
      options: {
        backgrounds: null,
        knobs: { addDecorator: false },
      },
    },
  ],
  stories: ['../src/stories/**/*.stories.(js|ts|tsx)'],
};
```

However this produces an error. Am I doing something wrong? I filed another issue about triconfig support for the CRA preset https://github.com/storybookjs/presets/issues/71 but I don't know whether it's the problem.

Here's the error for completeness:

```
ERROR in ./.storybook/generated-entry.js
Module build failed (from /Users/shilman/projects/baseline/storybook/node_modules/babel-loader/lib/index.js):
Error: Duplicate plugin/preset detected.
If you'd like to use two separate instances of a plugin,
they need separate names, e.g.

  plugins: [
    ['some-plugin', {}],
    ['some-plugin', {}, 'some unique name'],
  ]

Duplicates detected are:
[
  {
    "alias": "/Users/shilman/projects/baseline/storybook/node_modules/babel-plugin-named-asset-import/index.js",
    "options": {
      "loaderMap": {
        "svg": {
          "ReactComponent": "@svgr/webpack?-prettier,-svgo![path]"
        }
      }
    },
    "dirname": "/Users/shilman/projects/baseline/storybook/examples/cra-ts-essentials",
    "ownPass": false,
    "file": {
      "request": "/Users/shilman/projects/baseline/storybook/node_modules/babel-plugin-named-asset-import/index.js",
      "resolved": "/Users/shilman/projects/baseline/storybook/node_modules/babel-plugin-named-asset-import/index.js"
    }
  },
  {
    "alias": "/Users/shilman/projects/baseline/storybook/node_modules/babel-plugin-named-asset-import/index.js",
    "options": {
      "loaderMap": {
        "svg": {
          "ReactComponent": "@svgr/webpack?-svgo,+titleProp,+ref![path]"
        }
      }
    },
    "dirname": "/Users/shilman/projects/baseline/storybook/examples/cra-ts-essentials",
    "ownPass": false,
    "file": {
      "request": "/Users/shilman/projects/baseline/storybook/node_modules/babel-plugin-named-asset-import/index.js",
      "resolved": "/Users/shilman/projects/baseline/storybook/node_modules/babel-plugin-named-asset-import/index.js"
    }
  }
]
```

File: addons/essentials/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  } catch (err) {
+    logger.error(`Error reading package.json: ${err.message}`);
+  }
+}
+
+const warnIfInstalled = (addon: string, dependencies?: Record<string, string>) => {
+  if (dependencies) {
+    const version = dependencies[addon];
+    if (version) {
+      logger.warn(dedent`
```
Comment (zol) [inline addons/essentials/src/index.ts]: We should log a recommendation in the warning, e.g "we recommend you remove it as a dependency of your app and allow addon-essentials to include the right version instead".

**Comment by shilman** [comment]
@zol I updated config mechanism & README per your feedback and I think it's better and good to go!

The triconfig issue remains @ndelangen @tmeasday

**Review by zol — APPROVED** [review]
👍

**Pull request merged**